pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -366,9 +366,32 @@ func (c *Cluster) checkHealth(ctx context.Context) error {
 		}
 	}
 
+	// failover a mon if its IP is required to be updated due to change in hostNetwork settings
+	monName, updateMonIP := c.isMonIPUpdateRequiredForHostNetwork()
+	if updateMonIP {
+		c.failMon(len(c.ClusterInfo.Monitors), desiredMonCount, monName)
+	}
+
 	return nil
 }
 
+// verify if the hostnetwork setting has changed on a running cluster by comparing the
+// mon IP and the node IP.
+func (c *Cluster) isMonIPUpdateRequiredForHostNetwork() (string, bool) {
+	for _, mon := range c.ClusterInfo.Monitors {
+		if c.mapping != nil {
+			monNodeAddress := c.mapping.Schedule[mon.Name].Address
+			hostNetworkEnabled := c.spec.Network.IsHost()
+			if hostNetworkEnabled && !strings.Contains(mon.Endpoint, monNodeAddress) ||
```

Comment:
Its a long if statement can we have 2 variables for it

Reply:
Just two conditional statements. I think variables are not needed here.

Reply:
Atleast we should have brackets (hostNetworkEnabled && !strings.Contains(mon.Endpoint, monNodeAddress)) || (!hostNetworkEnabled && strings.Contains(mon.Endpoint, monNodeAddress))

Otherwise two variables looks good